src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs

```diff
@@ -400,49 +371,54 @@ fieldDeclaration.Parent is CompilationUnitSyntax &&
                 }
             }
 
-            if (objectCreationExpressionOpt.Initializer != null)
-            {
-                foreach (var expression in objectCreationExpressionOpt.Initializer.Expressions)
-                {
-                    if (expression is not AssignmentExpressionSyntax simpleAssignmentExpression)
-                    {
-                        continue;
-                    }
-
-                    if (simpleAssignmentExpression.Left is not SimpleNameSyntax name)
-                    {
-                        continue;
-                    }
+            var initializer = objectCreationExpressionOpt.Initializer;
+            AddPropertiesToInitialize(generateTypeServiceStateOptions, initializer);
+        }
 
-                    generateTypeServiceStateOptions.PropertiesToGenerate.Add(name);
-                }
-            }
+        // Check for `NewType t = new() { ... }` and initialize properties if present 
+        if (nameOrMemberAccessExpression.Parent is VariableDeclarationSyntax
+            {
+                Variables: [{ Initializer.Value: ImplicitObjectCreationExpressionSyntax { Initializer: { } implicitInitializer } }, ..]
```

Comment:
>Variables [](http://example.com/codeflow?start=16&length=9)

Dumb question: Is there any reason we choose the first one? Would it make sense to not enable if more than one variable (or if more than one, only enable if they have the same initialized properties)?

Reply:
Goal here is to not go to down the rabbit hole of potential cases to support.

Reply:
chatted offline, Cyrus is being lazy :)